### ISSUE TITLE
add more configurable options for pod template

### ISSUE THREAD

**Issue by wey-gu** [vesoft-inc/nebula-operator#35]
During the tests of MEG friends, it's found there could be a conflict between nebulaCluster pods and istio-proxy sidecar pods, in order to disable the sidecar, below annotation fields are needed.

![IMG_0546](https://user-images.githubusercontent.com/1651790/122916468-7fa16600-d38f-11eb-89a1-73b5fdf34683.JPG)

While for now, there is no such configuration interface to do so.

Is it feasible to add configurable options on this kind of k/v pairs via `values.yaml` ?

**Comment by veezhang** [comment]
@wey-gu  Oh, Good idea!

**Comment by veezhang** [comment]
@wey-gu It's fix in #36 

After merged, please configure `'nebula.podAnnotations.sidecar\.istio\.io/inject` with `false`.

Notes:

* update the operator images
* update helm repo
    ```shell
    # If you have already added it, please skip.
    $ helm repo add nebula-operator https://vesoft-inc.github.io/nebula-operator/charts
    $ helm repo update
    ```
* re-install operator with `--set 'image.nebulaOperator.imagePullPolicy'=Always`
    ```shell
    # for example
    # image.nebulaOperator.imagePullPolicy'=Always is used for updates the images
    $ helm uninstall nebula-operator --namespace nebula-operator-system
    $ kubectl delete crd nebulaclusters.apps.nebula-graph.io
    $ helm install nebula-operator ./charts/nebula-operator \          
        --namespace nebula-operator-system --create-namespace \
        --set 'image.nebulaOperator.imagePullPolicy'=Always
    ```

For examples:

```shell
$ export NEBULA_CLUSTER_NAME=nebula           # the name for nebula cluster
$ export NEBULA_CLUSTER_NAMESPACE=nebula      # the namespace you want to install the nebula cluster
$ export STORAGE_CLASS_NAME=standard          # the storage class for the nebula cluster

$ helm install "${NEBULA_CLUSTER_NAME}" nebula-operator/nebula-cluster \
    --namespace="${NEBULA_CLUSTER_NAMESPACE}" --create-namespace\
    --set nameOverride=${NEBULA_CLUSTER_NAME} \
    --set nebula.storageClassName="${STORAGE_CLASS_NAME}" \
    --set-string 'nebula.podAnnotations.sidecar\.istio\.io/inject='false

$ kubectl get pod -n ${NEBULA_CLUSTER_NAMESPACE} -l app.kubernetes.io/cluster=${NEBULA_CLUSTER_NAME} \
    -o custom-columns='NAME:.metadata.name,Inject:.metadata.annotations.sidecar\.istio\.io/inject'
NAME                Inject
nebula-graphd-0     false
nebula-graphd-1     false
nebula-metad-0      false
nebula-metad-1      false
nebula-metad-2      false
nebula-storaged-0   false
nebula-storaged-2   false
```

**Comment by wey-gu** [comment]
@veezhang wow, thanks!!!
